extensions/memory-operands-1/README.md

```diff
@@ -69,34 +69,42 @@ read or write in which case it is _unspecified_ behavior.
 
 The `D` bit is the less significant `MM` bit, and refers to the "direction" of the operation.
 
-Let `A` refer to the value of the register specified by `AAA`. Let `M` refer to the value in the memory
-location denoted by the address as described above. Then all operations using this format have the following operation:
+Let `A` refer to the value of the register specified by `AAA`. Let `M` refer to the address encoded
+as described above, and let `[M]` be the value at that location. 
+Then all operations using this format have the following operation:
 ```
 if D == 0:
   OP_L ← A
-  OP_R ← M
+  if operation == LEA:
+    OP_R ← M
+  else:
+    OP_R ← [M]
 else:
   if dst_is_input(operation):
-    OP_L ← M
+    OP_L ← [M]
   OP_R ← A
 RES ← operation(OP_L, OP_R)
 if stores_result(operation):
   if D == 0:
     A ← RES
   else:
-    M ← RES
+    [M] ← RES
 ```
 In plain language; when `D` is 0, the `dst` operand is a register and the `src` operand is a memory location.
-When `D` is 1, these are swapped. In all cases, the value in the memory location must be loaded from memory.
-When the `dst` operation is a memory location, the result of the operation must also be stored there.
+When `D` is 1, these are swapped. Usually, the value in the memory location must be loaded from memory.
+When the `dst` operation is a memory location, the result of the operation must also usually be stored there. However:
 
-If an operation uses an `ABM` byte to encode two operands, but only treats one of them as an input, then this
-behavior applies likewise to memory operations; memory must not be (visibly) loaded if the memory operand is not an input
-(e.g., as is the case for `mov m, r/i`). If the operation does not store its result, then memory must not be modified.
+- If an operation uses an `ABM` byte to encode two operands, but does not treat the memory one as input,
+    memory must not be (visibly) loaded. Note that this means MMIO address cannot be read from, even
+    if the result of the read is immediately discarded.
+- If the operation is `LEA`, memory must likewise not be (visibly) loaded. `LEA` is interested in the address itself.
+- If the operation does not store its result (e.g. `cmp m,r`), then memory must not be (visibly) written to.
+    This also includes MMIO addresses.
```

Comment:
(very redundant IMO: You can't do any valid store, visible or invisible, for operations that don't have a result)

Reply:
I agree that the note on cmp is unneeded.